controllers/build-tool-detector.go

```diff
@@ -38,26 +44,93 @@ const (
 	buildToolDetectorController = "BuildToolDetectorController"
 )
 
+type tokenRetriever struct {
+	authClient *client.Client
+	context    *app.ShowBuildToolDetectorContext
+}
+
+// TokenForService calls auth service to retrieve a token for an external service (ie: GitHub).
+func (tr *tokenRetriever) tokenForService(forService string) (*string, error) {
```

Comment:
We should try to keep our private and public methods organized so the code is a little easier to follow. Maybe let's move this function below where the other private methods are?